### PR TITLE
Add MekLocation tests

### PR DESCRIPTION
**Pull request by sixlettervariables**
This adds MekLocation unit tests and fixes tiny nits along the way.

TODO:
- [x] getXXXTime and getXXXDifficulty
- [x] needsFixing
- [x] updateConditionFromPart
- [x] fix
- [x] checkFixable
- [x] remove
- [x] checkScrappable
- [x] checkSalvagable
- [x] doMaintenanceDamage
- [x] getAllMods
- [x] getDetails
- [x] getDesc

### CODE REVIEW

**Comment by sixlettervariables** [comment]
I've only got the small tests left to do, but otherwise all of the major functionality of MekLocation is now tested.

File: MekHQ/src/mekhq/campaign/parts/MekLocation.java [hunk 1]
```diff
@@ -191,18 +192,27 @@ public boolean forQuad() {
...
+        return (part instanceof MekLocation)
+                && isSamePartType((MekLocation) part);
+    }
+
+    /**
+     * Gets a value indicating whether or not {@code other} is the same
+     * type of {@code MekLocation}.
+     * @param other A different {@code MekLocation}.
+     */
+    public boolean isSamePartType(MekLocation other) {
```
Comment (Windchild292) [inline MekHQ/src/mekhq/campaign/parts/MekLocation.java:204]: I concur on the confusion here, especially as MekLocation extends Part

File: MekHQ/src/mekhq/campaign/parts/MekLocation.java [hunk 2]
```diff
@@ -537,13 +579,33 @@ public String getDetails(boolean includeRepairDetails) {
...
+        }
+
         return toReturn;
     }
 
     @Override
     public void updateConditionFromPart() {
         if (null != unit) {
-            unit.getEntity().setInternal((int)Math.round(percent * unit.getEntity().getOInternal(loc)), loc);
+            unit.getEntity().setInternal((int)Math.round(getPercent() * unit.getEntity().getOInternal(loc)), loc);
```
Comment (Windchild292) [inline MekHQ/src/mekhq/campaign/parts/MekLocation.java:608]:
```suggestion
            unit.getEntity().setInternal((int) Math.round(getPercent() * unit.getEntity().getOInternal(loc)), loc);
```

File: MekHQ/src/mekhq/campaign/parts/MekLocation.java [hunk 3]
```diff
@@ -694,15 +745,19 @@ public String checkScrappable() {
         //otherwise you will get weirdness where armor and actuators are
         //still attached but everything else is scrapped
         //cant salvage torsos until arms and legs are gone
+        String limbName = " arm ";
+        if (forQuad) {
+            limbName = " front leg ";
+        }
```
Comment (Windchild292) [inline MekHQ/src/mekhq/campaign/parts/MekLocation.java:751]: Ternary candidate

**Review by Windchild292 — APPROVED** [review]

**Comment by Windchild292** [comment]
@sixlettervariables just noticed this was unmerged. A few comments open and a merge conflict to fix, then this should be good to merge

**Comment by sixlettervariables** [comment]
@Windchild292 addressed. There were some odd merge conflicts in un-changed parts of the code (Acquisition dialog), but I have those resolved.

**Review by Windchild292 — APPROVED** [review]

**Pull request merged**